code/modules/events/tram_malfunction.dm

```diff
@@ -0,0 +1,77 @@
+#define TRAM_MALFUNCTION_TIME_UPPER 420
+#define TRAM_MALFUNCTION_TIME_LOWER 240
+#define XING_STATE_GREEN 0
+
+/datum/round_event_control/tram_malfunction
+	name = "Tram Malfunction"
+	typepath = /datum/round_event/tram_malfunction
+	weight = 20
+	max_occurrences = 4
+	category = EVENT_CATEGORY_ENGINEERING
+
+// If manually triggered, check if there's a tram.
+/datum/round_event_control/tram_malfunction/admin_setup()
```

Comment:
You should check `can_spawn_event()` or whatever instead of copying and pasting.

Reply:
Yep, cleaned this up.